lib/srv/discovery/discovery.go

```diff
@@ -108,9 +108,43 @@ func New(ctx context.Context, cfg *Config) (*Server, error) {
 		cancelfn: cancelfn,
 	}
 
+	var (
+		ec2Matcher []services.AWSMatcher
+	)
+
+	for _, matcher := range cfg.AWSMatchers {
+	typeLoop:
+		for _, t := range matcher.Types {
+			for _, region := range matcher.Regions {
+				switch t {
+				case constants.AWSServiceTypeEC2:
```

Comment:
nit: I think that `switch t` condition can be evaluated before processing `for _, region := range matcher.Regions`  that should allow to remove `typeLoop:`  `continue typeLoop` label jumps.